### ISSUE TITLE
Multiple Params not found?

### ISSUE THREAD

**Issue by Mark-McCracken** [kamilmysliwiec/nest#25]
I can't seem to get a second parameter to be picked up by the decorators, I've got the following code:
```js
@Controller('player-activity')
export class PlayerActivityController {
    constructor(private playerActivityService: PlayerActivityService) {}

    @Get('/:localeID/:memberID')
    async activity (@Response() res, @Param('localeID') localeID, @Param('memberID') memberID) {
        console.log(`player-activity.controller: localeID: ${localeID}, memberID: ${memberID}`);
        const playerActivity: any[] = await this.playerActivityService.getActivity(+localeID, +memberID);
        res.status(HttpStatus.OK).json(playerActivity);
    }
}
```
and the first param item is fine, but the second is coming through as undefined.

<img width="1213" alt="screen shot 2017-05-04 at 23 24 27" src="https://cloud.githubusercontent.com/assets/14816091/25727712/bd9e23d0-3122-11e7-8763-f728dfbac9a2.png">
(ignore first log line )

Is this an issue or something I have done wrong?

**Comment by Mark-McCracken** [comment]
I think this is due to the following file

src/common/utils/decorators/route-params.decorator.ts
```js
import 'reflect-metadata';
import { ROUTE_ARGS_METADATA } from '../../constants';
import { RouteParamtypes } from '../../enums/route-paramtypes.enum';

export type ParamData = object | string | number;
export interface RouteParamsMetadata {
    [prop: number]: {
        index: number;
        data?: ParamData;
    };
}

const assignMetadata = (
    args: RouteParamsMetadata,
    paramtype: RouteParamtypes,
    index: number,
    data?: ParamData) => ({
    ...args,
    [paramtype]: { index, data },
});

const createRouteParamDecorator = (paramtype: RouteParamtypes) => {
    return (data?: ParamData): ParameterDecorator => (target, key, index) => {
        const args = Reflect.getMetadata(ROUTE_ARGS_METADATA, target, key) || {};
        Reflect.defineMetadata(
            ROUTE_ARGS_METADATA,
            assignMetadata(args, paramtype, index, data),
            target,
            key,
        );
    };
};

export const Request: () => ParameterDecorator = createRouteParamDecorator(RouteParamtypes.REQUEST);
export const Response: () => ParameterDecorator = createRouteParamDecorator(RouteParamtypes.RESPONSE);
export const Next: () => ParameterDecorator = createRouteParamDecorator(RouteParamtypes.NEXT);
export const Query: (property?: string) => ParameterDecorator = createRouteParamDecorator(RouteParamtypes.QUERY);
export const Body: (property?: string) => ParameterDecorator = createRouteParamDecorator(RouteParamtypes.BODY);
export const Param: (property?: string) => ParameterDecorator = createRouteParamDecorator(RouteParamtypes.PARAM);
export const Session: () => ParameterDecorator = createRouteParamDecorator(RouteParamtypes.SESSION);
export const Headers: (property?: string) => ParameterDecorator = createRouteParamDecorator(RouteParamtypes.HEADERS);
```

the createRouteParamDecorator function I think will be called multiple times when using the decorator twice.
The assignMetadata function is using [paramtype] as the key, however this key will not be unique when using enums.
This needs to be combined with the index, perhaps as a Map value I think, so that when then next decorator comes along with the same enum (Param or body), it doesn't cause a conflict with the existing key.

**Comment by Mark-McCracken** [comment]
perhaps this could solve it?

export interface RouteParamsMetadata {
    [prop: string]: {
        index: number;
        data?: ParamData;
    };
}

const assignMetadata = (
    args: RouteParamsMetadata,
    paramtype: RouteParamtypes,
    index: number,
    data?: ParamData) => ({
    ...args,
    [`paramtype:${paramtype}, index:${index}`]: { index, data },
});

**Comment by kamilmysliwiec** [comment]
Hi @Mark-McCracken,
Thanks for the report.

Issue is fixed in 1.0.2 version. Please, update your package :)
